### DISCUSSION TITLE
爬梯子

### DISCUSSION THREAD

**Discussion by WhiteCloud0106** [Lost-MSth/Arcaea-server#142] [Q&A]
想問梯子的解鎖條件設定
要怎麼樣才可以像world extend那樣沒有條件就可以解梯子
試了很多方法還是會在點它的時候顯示"未解鎖???"

**Comment by Lost-MSth** [comment]
先看看这个 https://www.bilibili.com/read/cv22336290/

没懂您想要的是什么效果，无条件不就是什么都不设置吗

**Reply by WhiteCloud0106** [reply]
的確是這樣設置了沒錯，但是遊戲內不知道為什麼進不去
![螢幕擷取畫面 2023-12-26 205436](https://github.com/Lost-MSth/Arcaea-server/assets/94116682/e23bf932-d78e-43a9-b729-f96e6b0c5291)
![download](https://github.com/Lost-MSth/Arcaea-server/assets/94116682/daf11a35-0df1-4a3b-bf27-282d166b3691)
還是有哪邊不對

**Reply by Lost-MSth** [reply]
啊？这是我写进去的测试图吗，那应该是可以直接进入的啊，难道壳子有问题吗

**Reply by WhiteCloud0106** [reply]
這是我另外寫的一個測試限時的圖，但感覺不是殼子的問題
有另一張普通地圖是進得去的，但目前也只有那張進得去，其他張全都是像第二張圖那樣顯示未解鎖

**Reply by Lost-MSth** [reply]
这可太怪了哦，那张能够进入的地图的差别在哪里呢，只是不限时吗

**Reply by WhiteCloud0106** [reply]
我好像找到問題了，我剛剛把你原本自帶的幾個測試圖丟進去也無法作用，然後我把裡面“require_localunlock_songid”隨便加上一首歌，結果就可以用了🤯
我猜應該是這個require一定要有東西讓他偵測的樣子，原先可以進去的那個地圖這裡也是有填了某首曲子
![Screenshot_20231226_223119](https://github.com/Lost-MSth/Arcaea-server/assets/94116682/a1518682-2cb9-436a-8b89-277f1f9a0e35)

**Reply by Lost-MSth** [reply]
我刚刚测试过了，并没有问题，我写的 "test3" 图可以正常解锁，不存在必须写一个 require 的要求

**Reply by WhiteCloud0106** [reply]
可是我這邊不行欸😨
![Screenshot_20231226_231028](https://github.com/Lost-MSth/Arcaea-server/assets/94116682/2169093c-7f00-4b0a-8987-225ec8afaa09)

**Comment by WhiteCloud0106** [comment]
再問一下如果限時活動結束之後要讓玩家看不到該怎麼處理

**Reply by Lost-MSth** [reply]
写个脚本定时删了图再刷新地图缓存吧

官服是会自动删的吗，至少没有别的操作的话，只会显示地图不可用

**Reply by WhiteCloud0106** [reply]
刷新地圖緩存是甚麼意思🤔

**Reply by Lost-MSth** [reply]
重开服务端，或者，因为我没写方便的接口，所以只能调用 `core/world.py` 下的前两个函数的缓存刷新方法

**Reply by WhiteCloud0106** [reply]
想問一下後面提到的方法具體如何使用呢?
服務器資料太多，每次重開都需要超過五分鐘了(((

**Reply by Lost-MSth** [reply]
看了下，就一个函数 `MapParser.get_world_info` 有缓存，调用它的 `cache_clear()` 方法

我没写但是你可以看看 `core/operation.py` 下面的操作，这里面的类都可以被 API 接口调用

唔，加入到以后得更新计划中（）

另外，重开后歌曲数据要五分钟才能读完？有点离谱了，难不成是机械盘？

**Reply by WhiteCloud0106** [reply]
估計可能是，用的雲服務器
加上裡面1000多首歌曲，每次開都等挺久的😂
